javaparser-core/src/main/java/com/github/javaparser/ast/Node.java

```diff
@@ -431,7 +473,18 @@ public boolean remove() {
                     if (optSetter.isPresent()) {
                         try {
                             Object resultRaw = method.invoke(parentNode);
-                            Node result = isOptionalAssignableFrom(method.getGenericReturnType(), this.getClass()) ? (Node)((Optional)resultRaw).get(): (Node) resultRaw;
+                            Node result;
+                            if (isOptionalAssignableFrom(method.getGenericReturnType(), this.getClass())) {
+                                Optional optionalResultRaw = (Optional) resultRaw;
+                                if (optionalResultRaw.isPresent()) {
+                                    Object o = optionalResultRaw.get();
+                                    if (Node.class.isAssignableFrom(o.getClass())) {
```

Comment:
we do not need this check because anyway we are comparing by identity later, so no need to cast